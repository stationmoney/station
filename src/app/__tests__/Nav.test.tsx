import React from "react"
import { render } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"
import { RecoilRoot } from "recoil"
import Nav from "app/sections/Nav"

function renderComponent() {
  return render(
    <MemoryRouter>
      <RecoilRoot>
        <Nav />
      </RecoilRoot>
    </MemoryRouter>
  )
}

jest.mock("lottie-react", () => ({
  Lottie: jest.fn(),
}))

jest.mock("../../data/settings/Theme", () => {
  const mockUseThemeFavicon = () => "/static/media/favicon.1e08d51d.svg"

  return {
    useThemeFavicon: mockUseThemeFavicon,
  }
})

describe("Nav component matches snapshots", () => {
  it("matches original component", () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })
})
