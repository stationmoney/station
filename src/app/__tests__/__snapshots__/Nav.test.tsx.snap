// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshots matches original component 1`] = `
<DocumentFragment>
  <nav>
    <header
      class="header"
    >
      <div
        class="item logo"
      >
        <img
          alt="Station"
          src="/static/media/favicon.1e08d51d.svg"
        />
         
        <strong
          class="title"
        >
          Station
        </strong>
      </div>
    </header>
    <a
      aria-current="page"
      class="item link active"
      data-id="/"
      href="/"
    >
      <svg
        height="20"
        width="20"
      >
        Swap.svg
      </svg>
      Swap
    </a>
    <a
      class="item link"
      data-id="/history"
      href="/history"
    >
      <svg
        height="20"
        width="20"
      >
        History.svg
      </svg>
      History
    </a>
    <a
      class="item link"
      data-id="/stake"
      href="/stake"
    >
      <svg
        height="20"
        width="20"
      >
        Stake.svg
      </svg>
      Stake
    </a>
    <a
      class="item link"
      data-id="/gov"
      href="/gov"
    >
      <svg
        height="20"
        width="20"
      >
        Governance.svg
      </svg>
      Governance
    </a>
    <a
      class="item link"
      data-id="/contract"
      href="/contract"
    >
      <svg
        height="20"
        width="20"
      >
        Contract.svg
      </svg>
      Contract
    </a>
  </nav>
</DocumentFragment>
`;
